DigitalLearningSolutions.Data/DataServices/SelfAssessmentDataService/FilteredDataService.cs

```diff
@@ -0,0 +1,52 @@
+﻿namespace DigitalLearningSolutions.Data.DataServices.SelfAssessmentDataService
+{
+    using System.Collections.Generic;
+    using System.Data;
+    using Dapper;
+    using DigitalLearningSolutions.Data.Models.External.Filtered;
+
+    public partial class SelfAssessmentDataService
+    {
+        public Profile? GetFilteredProfileForCandidateById(int selfAssessmentId, int candidateId)
+        {
+            return connection.QueryFirstOrDefault<Profile>(
+                "GetFilteredProfileForCandidate",
+                new { selfAssessmentId, candidateId },
+                commandType: CommandType.StoredProcedure
+            );
+        }
+
+        public IEnumerable<Goal> GetFilteredGoalsForCandidateId(int selfAssessmentId, int candidateId)
+        {
+            return connection.Query<Goal>(
+                "GetFilteredCompetencyResponsesForCandidate",
+                new { selfAssessmentId, candidateId },
+                commandType: CommandType.StoredProcedure
+            );
+        }
+
+        public void LogAssetLaunch(int candidateId, int selfAssessmentId, LearningAsset learningAsset)
+        {
+            connection.Execute(
+                "UpdateFilteredLearningActivity",
+                new
+                {
+                    FilteredAssetID = learningAsset.Id,
+                    learningAsset.Title,
+                    learningAsset.Description,
+                    learningAsset.DirectUrl,
+                    Type = learningAsset.TypeLabel,
+                    Provider = learningAsset.Provider.Name,

```

Comment:
Warning about learningAsset.Provider being potentially null. I doubt we are familiar enough with this to know what situtations, if any, this could be null. Do you think it's safe enough to use the null conditional operator here?

Reply:
I know nothing about the stored procedure here and what values it gets. This learning asset seems to come from the Filtered API so I think it's safer to leave as is.